pixelart/pixelart.py

```diff
@@ -149,25 +149,38 @@ def snake_mode(document: vp.Document, img: np.ndarray, colors: np.ndarray, pen_w
         document.add(lc, col_idx)
 
 
+def dot_mode(document: vp.Document, img: np.ndarray, colors: np.ndarray, pen_width: float):
+    for col_idx, color in enumerate(colors, start=1):
+        indice_i, indice_j = np.nonzero(
+            np.all(img[:, :, 0:3] == color, axis=2) & (img[:, :, 3] == 255)
+        )
+
+        paths = [vp.circle(i, j, pen_width * 0.05)
+                 for i, j in zip(indice_j, indice_i)]
```

Comment:
This should read as follows, for correctness and clarity:
```
paths = [vp.circle(j * pen_width, i * pen_width, pen_width * 0.05)
                  for j, i in zip(indice_j, indice_i)]
```

In particular, the `* pen_width` is needed such that the grid is indeed scaled by the pen width.